### PR TITLE
Send productImpression events to PixelManager

### PR DESCRIPTION
**Pull request by italo-batista**
#### What is the purpose of this pull request?
Sending productImpression event to Pixel Manager with the purpose of usage in google-analytics.

A productImpression is an event defined as the moment a user sees a product in a list of products. It occurs before productClick and productDetail events.

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [X] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

**Workspace**: https://aguaviva--storecomponents.myvtex.com/

Related: [pixel-manager#19](https://github.com/vtex/pixel-manager/pull/19)

### CODE REVIEW

File: react/ShelfContent.js [hunk 1]
```diff
@@ -107,6 +116,12 @@ class ShelfContent extends Component {
 
     const { currentSlide } = this.state
 
+    if (firstRender) {
+      productList.slice(0, maxItems).forEach((product, index) => {
```
Comment (klzns) [inline react/ShelfContent.js]: Do not add side-effects to render.

Move this to `componentDidMount`

File: react/ShelfContent.js [hunk 2]
```diff
@@ -94,6 +96,19 @@ class ShelfContent extends Component {
     )
   }
 
+  pushPixelProductsImpressions = () => {
+    const { products } = this.props
+
+    products.forEach((product, index) => {
+      this.props.push({
+        event: 'productImpression',
+        list: 'Home Page Shelf',
```
Comment (klzns) [inline react/ShelfContent.js]: Since this is not a required field and the shelf is not always the home shelf, I think we could remove this for now.

Comment (italo-batista) [reply react/ShelfContent.js]: Or mantaining just _Shelf_. Could be strategically interesting.

File: react/ShelfContent.js [hunk 3]
```diff
@@ -94,6 +96,19 @@ class ShelfContent extends Component {
...
 
+  pushPixelProductsImpressions = () => {
+    const { products } = this.props
+    if (products)
+      products.forEach((product, index) => {
+        this.props.push({
+          event: 'productImpression',
+          list: 'Shelf',
+          position: index + 1,
+          product,
```
Comment (klzns) [inline react/ShelfContent.js]: Create some way to send the same product of the other events, example of missing property `selectedSku`.

**Pull request merged**